lightkurve/interact.py

```diff
@@ -0,0 +1,394 @@
+                        color_mapper=color_mapper)
+
+    # The colorbar will update with the screen stretch slider
+    # The colorbar margin increases as the length of the tick labels grows.
+    # This colorbar share of the plot window grows, shrinking plot area.
+    # This effect is known, some workarounds might work to fix the plot area:
+    # https://github.com/bokeh/bokeh/issues/5186
+    color_bar = ColorBar(color_mapper=color_mapper,ticker=LogTicker(),
+                         label_standoff=-10, border_line_color=None, location=(0,0),
+                         background_fill_color='whitesmoke',major_label_text_align = 'left',
+                         major_label_text_baseline = 'middle',title = 'e/s', margin = 0)
+    fig.add_layout(color_bar, 'right')
+    color_bar.formatter = PrintfTickFormatter(format="%14u")#NumeralTickFormatter(format='      0,0')
+
+    fig.rect('xx', 'yy', 1, 1, source=tpf_source, fill_color='gray',
+                      fill_alpha=0.4, line_color='white')
+
+    # Lasso Select apparently does not work with boxes/quads/rect.
+    # See https://github.com/bokeh/bokeh/issues/964
+    #fig.add_tools(LassoSelectTool(renderers=[pixels], select_every_mousemove=False)
+
+    return (fig, fig_dat,
+            {'pedestal':pedestal, 'vlo':vlo, 'lo':lo, 'med':med,
+             'hi':hi, 'vhi':vhi, 'vstep':vstep})
+
+def pixel_selector_standalone(tpf, notebook_url='localhost:8888'):
+    """Display an interactive IPython Notebook widget to select pixel masks.
+
+    The widget will show both the lightcurve and pixel data.  The pixel data
+    supports pixel selection via Bokeh tap and box select tools in an
+    interactive javascript user interface.
+
+    Note: at this time, this feature only works inside an active Jupyter
+    Notebook, and tends to be too slow when more than ~30,000 cadences
+    are contained in the TPF (e.g. short cadence data).
+
+    Parameters
+    ----------
+    tpf: `lightkurve.TargetPixelFile` instance
+        Target Pixel File to interact with
+    notebook_url: str
+        Location of the Jupyter notebook page (default: "localhost:8888")
+
+        When showing Bokeh applications, the Bokeh server must be
+        explicitly configured to allow connections originating from
+        different URLs. This parameter defaults to the standard notebook
+        host and port. If you are running on a different location, you
+        will need to supply this value for the application to display
+        properly. If no protocol is supplied in the URL, e.g. if it is
+        of the form "localhost:8888", then "http" will be used.
+    """
+
+    lc = tpf.to_lightcurve(aperture_mask=tpf.pipeline_mask)
+
+    n_pixels = tpf.flux[0, :, :].size
+    pixel_index_array = np.arange(0, n_pixels, 1, dtype=int).reshape(tpf.flux[0, :, :].shape)
+
+    # Bokeh cannot handle many data points
+    # https://github.com/bokeh/bokeh/issues/7490
+    if len(lc.cadenceno) > 30000:
```

Comment:
To be future-proof, I recommend we do not hard-code the `30000` limit and make this an optional parameter of the interact function instead?  The default of that parameter can be 30000.